### ISSUE TITLE
MockK 1.10.4 is no longer compatible with Kotlin 1.3.*

### ISSUE THREAD

**Issue by tmohme** [mockk/mockk#553]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### Expected Behavior

Under "Kotlin version support" the website states "From version 1.10.0 MockK does not support Kotlin 1.2.\*".
However, it does not state that from 1.10.4 MockK does not support Kotlin 1.3.\*

### Current Behavior

During test execution with MockK 1.10.4 & Kotlin 1.3.72 I get the following Exception:
```
Caused by: java.lang.NoSuchMethodError: 'void kotlin.jvm.internal.FunctionReferenceImpl.<init>(int, java.lang.Class, java.lang.String, java.lang.String, int)'
	at io.mockk.impl.JvmMockKGateway$callRecorderFactories$3.<init>(JvmMockKGateway.kt)
	at io.mockk.impl.JvmMockKGateway$callRecorderFactories$3.<clinit>(JvmMockKGateway.kt)
	at io.mockk.impl.JvmMockKGateway.<init>(JvmMockKGateway.kt:122)
	at io.mockk.impl.JvmMockKGateway.<clinit>(JvmMockKGateway.kt:173)
```

#### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Configure project to use Kotlin 1.3.\*
2. Configure Project to use MockK 1.10.4
3. Run tests using MockK

#### Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* MockK version: 1.10.4
* OS: macOS 10.15.7
* Kotlin version: 1.3.72
* JDK version: 
* JUnit version: junit-platform 1.7.0, junit-jupiter-api 5.7.0
* Type of test: unit test

#### Stack trace

```
net.jqwik.api.JqwikException: Cannot create instance of class 'class org.acme.ClassUnderTest$ReportText'

	at net.jqwik.engine.execution.PropertyTaskCreator.createTestInstance(PropertyTaskCreator.java:109)
	at net.jqwik.engine.execution.PropertyTaskCreator.lambda$createTask$1(PropertyTaskCreator.java:37)
	at net.jqwik.engine.execution.pipeline.ExecutionTask$1.lambda$execute$0(ExecutionTask.java:31)
	at net.jqwik.engine.execution.lifecycle.CurrentTestDescriptor.runWithDescriptor(CurrentTestDescriptor.java:27)
	at net.jqwik.engine.execution.pipeline.ExecutionTask$1.execute(ExecutionTask.java:31)
	at net.jqwik.engine.execution.pipeline.ExecutionPipeline.runToTermination(ExecutionPipeline.java:82)
	at net.jqwik.engine.execution.JqwikExecutor.execute(JqwikExecutor.java:46)
	at net.jqwik.engine.JqwikTestEngine.executeTests(JqwikTestEngine.java:77)
	at net.jqwik.engine.JqwikTestEngine.execute(JqwikTestEngine.java:61)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:71)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:220)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:53)
Caused by: java.lang.NoSuchMethodError: 'void kotlin.jvm.internal.FunctionReferenceImpl.<init>(int, java.lang.Class, java.lang.String, java.lang.String, int)'
	at io.mockk.impl.JvmMockKGateway$callRecorderFactories$3.<init>(JvmMockKGateway.kt)
	at io.mockk.impl.JvmMockKGateway$callRecorderFactories$3.<clinit>(JvmMockKGateway.kt)
	at io.mockk.impl.JvmMockKGateway.<init>(JvmMockKGateway.kt:122)
	at io.mockk.impl.JvmMockKGateway.<clinit>(JvmMockKGateway.kt:173)
	at org.acme.ClassUnderTest$ReportText.<init>(ReportBuilderImplProperties.kt:1092)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at net.jqwik.engine.support.JqwikReflectionSupport.newInstance(JqwikReflectionSupport.java:100)
	at net.jqwik.engine.execution.TestInstanceCreator.newInstance(TestInstanceCreator.java:60)
	at net.jqwik.engine.execution.TestInstanceCreator.create(TestInstanceCreator.java:49)
	at net.jqwik.engine.execution.TestInstanceCreator.create(TestInstanceCreator.java:32)
	at net.jqwik.engine.execution.PropertyTaskCreator.createTestInstanceWithResolvedParameters(PropertyTaskCreator.java:118)
	at net.jqwik.engine.execution.PropertyTaskCreator.lambda$createTestInstance$2(PropertyTaskCreator.java:95)
	at net.jqwik.engine.execution.lifecycle.CurrentTestDescriptor.runWithDescriptor(CurrentTestDescriptor.java:27)
	at net.jqwik.engine.execution.PropertyTaskCreator.lambda$createTestInstance$3(PropertyTaskCreator.java:93)
	at java.base/java.util.Optional.map(Optional.java:265)
	at net.jqwik.engine.execution.PropertyTaskCreator.createTestInstance(PropertyTaskCreator.java:84)
	... 19 more
```

**Comment by Raibaz** [comment]
This is still because of [this](https://youtrack.jetbrains.com/issue/KT-39389).

I thought [this](https://github.com/mockk/mockk/commit/26569b1c462349b75ee5fd40884e678a49b8e8ab) commit was enough to achieve backwards compatibility with 1.3.* while compiling with 1.4, but it looks like it's not that way.

The next version will be compiled with the kotlinOption `apiVersion = '1.3'` which will ensure compatibility with 1.3.*.

On a positive note, v1.10.3 and v1.10.4 contain very little updates, so not being able to upgrade from 1.10.2 should not be a big deal.

**Comment by Raibaz** [comment]
Released v1.10.5, I ended up reverting back to Kotlin 1.3.72, will upgrade again to 1.4 when its adoption will be higher and backwards compatibility issues will not be widespread.
